subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -585,16 +594,29 @@ u16_t lwm2m_get_rd_data(u8_t *client_data, u16_t size)
 	u16_t pos = 0;
 	int len;
 
+	/* Add resource-type/content-type to the registration message */
+	memcpy(client_data, REG_PREFACE, sizeof(REG_PREFACE) - 1);
+	pos += sizeof(REG_PREFACE) - 1;
+
 	SYS_SLIST_FOR_EACH_CONTAINER(&engine_obj_list, obj, node) {
-		len = snprintf(temp, sizeof(temp), "%s</%u>",
-			       (pos > 0) ? "," : "", obj->obj_id);
-		if (pos + len >= size) {
-			/* full buffer -- exit loop */
-			break;
+		/* Security obj MUST NOT be part of registration message */
+		if (obj->obj_id == LWM2M_OBJECT_SECURITY_ID) {
+			continue;
 		}
 
-		memcpy(&client_data[pos], temp, len);
-		pos += len;
+		/* Only report <OBJ_ID> when no instance available */
+		if (obj->instance_count == 0) {
```

Comment:
The modification is to avoid sending object "</1>" when object instance "</1/0>" is available and reduce the payload size.

As you just mentioned, when object instances are already available on the client at registration time. 

> </1/0>,</1/1>,</2/0>,</2/1>,</2/2>,</2/3>,</2/4>,</3/0>,</4/0>,</5>"
 
You can see that tags such as </1>, </2>, </3>, </4> are ignored.
And </5> is appended because it doesn't have object instances.